controllers/controllerconfig_controller.go

```diff
@@ -85,14 +89,49 @@ func (r *ControllerConfigReconciler) Reconcile(ctx context.Context, req ctrl.Req
 		return ctrl.Result{}, fmt.Errorf("unable to create signer from private key secret: %w", err)
 	}
 
+	caBundle := ""
+	for _, bundle := range cc.Spec.ImageRegistryBundleUserData {
+		// Accumulate the certs into a single bundle, adding a comment with the image repo name for observability
+		caBundle += fmt.Sprintf("# %s\n%s\n\n", strings.ReplaceAll(bundle.File, "..", ":"), bundle.Data)
+	}
+	for _, bundle := range cc.Spec.ImageRegistryBundleData {
+		// Accumulate the certs into a single bundle, adding a comment with the image repo name for observability
+		caBundle += fmt.Sprintf("# %s\n%s\n\n", strings.ReplaceAll(bundle.File, "..", ":"), bundle.Data)
+	}
+	// merge registry certs with proxy certs, if enabled
+	if cluster.IsProxyEnabled() {
+		proxyCA := &core.ConfigMap{}
+		if err := r.client.Get(context.TODO(), types.NamespacedName{Namespace: r.watchNamespace,
+			Name: certificates.ProxyCertsConfigMap}, proxyCA); err != nil {
+			return ctrl.Result{}, fmt.Errorf("unable to get ConfigMap %s: %w", certificates.ProxyCertsConfigMap, err)
+		}
+		caBundle += proxyCA.Data[certificates.CABundleKey]
+	}
```

Comment:
Please break this out into a separate function

Reply:
Having both this and `SyncTrustedCABundle` is confusing to me.
Will `SyncTrustedCABundle` not overwrite the certs from ImageRegistryBundleUserData + ImageRegistryBundleData? Shouldn't all of this be in that function?

Reply:
updated PTAL if it's clear now.